File: python/diamond.py
# https://www.codewars.com/kata/5503013e34137eeeaa001648

def diamond(n):
    if n < 0 or not n % 2:
        return None
    elif n == 1:
        return '*\n'
    len = n // 2
    side = '\n'.join(f"{' ' * (len - i)}{'*' * ((i + 1) * 2 - 1)}" for i in range(len)) + '\n'
    return side + '*' * n + '\n'.join(side.split('\n')[::-1]) + '\n'
  
expected =  " *\n"
expected += "***\n"
expected += " *\n"
test.assert_equals(diamond(1), "*\n")
test.assert_equals(diamond(2), None)
test.assert_equals(diamond(3), expected)
test.assert_equals(diamond(5), "  *\n ***\n*****\n ***\n  *\n")
test.assert_equals(diamond(0), None)
test.assert_equals(diamond(-3), None)

⬇️⬇️

"""Example (n = 5)：
  *
 ***
*****
 ***
  *     
"""

File: Javascript/diamond.js
// https://www.codewars.com/kata/5503013e34137eeeaa001648

function diamond(n){
  if (n < 0 || n % 2 == 0) return null
  else if(n == 1) return '*\n'
  let len = Math.floor(n / 2)
  let side = Array.from({length: len}, (_, i) => `${' '.repeat(len - i)}${'*'.repeat((i + 1) * 2 - 1)}`).join('\n')
  return [side, '*'.repeat(n), side.split`\n`.reverse().join`\n`].join`\n` + '\n' 
}

describe( "diamond()", function(){

  Test.assertEquals(diamond(1), "*\n")
  Test.assertEquals(diamond(3), " *\n***\n *\n")
  Test.assertEquals(diamond(5), "  *\n ***\n*****\n ***\n  *\n")
  Test.assertEquals(diamond(2), null)
  Test.assertEquals(diamond(-3), null)
  Test.assertEquals(diamond(0), null)
});

⬇️⬇️

/* Example (n = 5)：
  *
 ***
*****
 ***
  *     
*/
